app/src/main/java/nu/yona/app/api/manager/impl/ActivityManagerImpl.java

```diff
@@ -308,57 +307,56 @@ public void postActivityToDB(String applicationName, Date startDate, Date endDat
 		}
 	}
 
+	/**
+	 * App activity Business
+	 */
+
+	@Override
+	public void postAllDBActivities()
+	{
+		postNextBatchOfAppActivities();
+	}
+
+	private void postNextBatchOfAppActivities()
+	{ // recursive function posts until all activities are posted to the server .
+		List<Activity> activityList = activityTrackerDAO.getActivities();
+		if (activityList == null || activityList.isEmpty())
+		{
+			isSyncAPICallDone = true;
+			return;
+		}
+		AppActivity appActivity = new AppActivity();
+		appActivity.setDeviceDateTime(DateUtility.getLongFormatDate(new Date()));
+		appActivity.setActivities(activityList);
+		postActivityOnServer(appActivity);
+	}
+
 	private boolean isSyncAPICallDone = true;
 
-	private void postActivityOnServer(AppActivity activity, boolean fromDB)
+	private void postActivityOnServer(AppActivity activity)
 	{
-		Logger.logi("postActivityOnServer", "isSyncAPICallDone: " + isSyncAPICallDone);
-		User user = getSharedAppDataState().getUser();
 		if (!isSyncAPICallDone)
 		{
 			return;
 		}
-		isSyncAPICallDone = false;
-		DataLoadListenerImpl dataLoadListenerImpl = new DataLoadListenerImpl((result) -> handlePostAppActivityOnSuccess(fromDB), (result) -> handlePostAppActivityOnFailure(fromDB, activity), null);
+		User user = getSharedAppDataState().getUser();
+		DataLoadListenerImpl dataLoadListenerImpl = new DataLoadListenerImpl((result) -> handlePostAppActivityOnSuccess(), (result) -> handlePostAppActivityOnFailure(activity), null);
 		activityNetwork.postAppActivity(user.getPostDeviceAppActivityLink(), YonaApplication.getEventChangeManager().getSharedPreference().getYonaPassword(), activity, dataLoadListenerImpl);
-
 	}
 
-	private Object handlePostAppActivityOnSuccess(Boolean fromDB)
+	private Object handlePostAppActivityOnSuccess()
 	{
-		//on success nothing to do, as it is posted on server. #JIRA_1022
-		if (fromDB)
-		{
-			activityTrackerDAO.clearActivities();
-		}
-		isSyncAPICallDone = true;
+		activityTrackerDAO.clearActivities();
+		postNextBatchOfAppActivities();
 		return null; // Dummy return value, to allow use as data load handler
 	}
 
-	private Object handlePostAppActivityOnFailure(Boolean fromDB, AppActivity activity)
+	private Object handlePostAppActivityOnFailure(AppActivity activity)
```

Comment:
Why is this error not being reported?

Reply:
User can not take any action on this as this happens in background. 
This can be reported to Crashlytics as a custom non-fatal.

Reply:
Indeed. And with that it also goes to ``LogCat``

Reply:
Can we log AppActivity argument posted to sever, in failure case ?

Reply:
The count, you could do, but not the entire set of them, as that could be as many as 200 items.

Reply:
Count is already logged in other file. 
As of now the Tag is style in the old style. I will update them to once your other [PR](https://github.com/yonadev/yona-app-android/pull/582) is merged.